resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/KubernetesSuite.scala

```diff
@@ -364,8 +308,12 @@ private[spark] class KubernetesSuite extends SparkFunSuite with BeforeAndAfter {
         .pods
         .withLabel("spark-app-name", "spark-file-existence-test")
         .watch(watch)) { _ =>
-      SparkSubmit.main(args)
```

Comment:
It's beneficial to test using `SparkSubmit` if we expect `SparkSubmit` to process its arguments in a way that our `Client` doesn't expect, but for the most part otherwise we should call the `Client` directly.